cmis/models/cmis_backend.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+# © 2014-2015 Savoir-faire Linux (<http://www.savoirfairelinux.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import cmislib.exceptions
+
+from openerp import api, fields, models
+from openerp.exceptions import Warning
+from openerp.tools.translate import _
+from openerp.addons.connector.connector import ConnectorEnvironment
+from openerp.addons.connector.session import ConnectorSession
+from ..unit.backend_adapter import CmisAdapter
+from ..exceptions import CMISError
+
+
+class CmisBackend(models.Model):
+    _name = 'cmis.backend'
+    _description = 'CMIS Backend'
+    _inherit = 'connector.backend'
+
+    _backend_type = 'cmis'
+
+    version = fields.Selection(
+        selection=[('1.0', '1.0')], required=True)
+    location = fields.Char(
+        required=True)
+    username = fields.Char(
+        required=True)
+    password = fields.Char(
+        required=True)
+    initial_directory_read = fields.Char(
+        'Initial directory for reading', required=True, default='/')
+    initial_directory_write = fields.Char(
+        'Initial directory for writing', required=True, default='/')
+    browsing_ok = fields.Boolean(
+        'Allow browsing this backend')
+    storing_ok = fields.Boolean(
+        'Allow storing in this backend')
+
+    @api.multi
+    def _get_base_adapter(self):
+        """
+        Get an adapter to test the backend connection
+        """
+        self.ensure_one()
+        session = ConnectorSession.from_env(self.env)
+        environment = ConnectorEnvironment(self, session, None)
+        return CmisAdapter(environment)
+
+    @api.multi
+    def check_auth(self):
+        """ Check the authentication with DMS """
+        self.ensure_one()
+        adapter = self._get_base_adapter()
+        return adapter._auth(self)
+
+    @api.multi
+    def check_directory_of_write(self):
+        """Check access right to write from the path"""
+        datas_fname = 'testdoc'
+        for this in self:
+            # login with the cmis account
+            repo = this.check_auth()
```

Comment:
@dufresnedavid The method doesn't create the directory if it doesn't exist. It only create a document to check if Odoo is able to write into the specified directory.